### DISCUSSION TITLE
HTTP GET and  POST

### DISCUSSION THREAD

**Discussion by miyamotodan** [comunica/comunica#1315] [Q&A]
Hi, some endopints do not allow POST calls but only GET calls, is there a way to force Comunica to use HTTP GET?

thanks

**Comment by rubensworks** [comment]
That's possible, but it requires configuring your own comunica engine and enabling this flag: https://github.com/comunica/comunica/blob/master/packages/actor-query-source-identify-hypermedia-sparql/lib/ActorQuerySourceIdentifyHypermediaSparql.ts#L71-L75 (`forceHttpGet`)

We should probably expose this flag as a context option at some point, which would avoid the need to configure your own engine (PRs for this are definitely welcome!).

That being said, I highly recommend configuring your endpoint to support POST. Sending large queries by GET can lead to problems where URLs become too long and get rejected by the endpoint. (this is the reason we default to POST)

**Reply by miyamotodan** [reply]
thanks... so now I have to follow this guide to enable the flag?
[https://comunica.dev/docs/modify/getting_started/custom_config_app/](https://comunica.dev/docs/modify/getting_started/custom_config_app/)

**Reply by rubensworks** [reply]
Yes, that's correct.
